### DISCUSSION TITLE
Missing of edge attributes in networkDynamic

### DISCUSSION THREAD

**Discussion by LettieWyq** [statnet/tergm#118] [Q&A]
Hi,
I encountered a problem while preparing the dynamic network list.
I was trying to fit a TERGM model with both vertex attributes and edge attributes(value).
I firstly constructed a list of networks with 4 periods.
`SpNetList <- list(Sp2019,Sp2020,Sp2021,Sp2022)`
The attributes were all correctly specified in the list.

> Vertex attribute names: 
>     CPI FINDEV GDPPC GGDP vertex.names 
> Edge attribute names: 
>     FIN TRA 

However, when I transformed the SpNetlist into networkDynamic, the edge attributes were missing (both were time-varying matrix):
`SpDy <- networkDynamic(network.list = SpNetList)`
> Vertex attribute names: 
>     active CPI FINDEV GDPPC GGDP vertex.names 
>  Edge attribute names: 
>     active 

Very much appreciate,
Yuqing W

**Comment by mbojan** [comment]
Thank you for reaching out. Can you provide a reprex?

**Reply by LettieWyq** [reply]
Yes! I omit the process of importing data, here is how I set attributes of the following networks:

> SpNetList <- list(Sp2019,Sp2020,Sp2021,Sp2022)
> 
> ### vertex attributes
> for(i in 1:length(SpNetList)){
>   for(feature in colnames(GGDP)){
>     SpNetList[[i]] <- set.vertex.attribute(SpNetList[[i]], "GGDP", GGDP[, i])
>   }
>   for(feature in colnames(GDPPC)){
>     SpNetList[[i]] <- set.vertex.attribute(SpNetList[[i]], "GDPPC", GDPPC[, i])
>   }
>   for(feature in colnames(CPI)){
>     SpNetList[[i]] <- set.vertex.attribute(SpNetList[[i]], "CPI", CPI[, i])
>   }
> }
> 
> ### edge attributes
> SpNetList <- lapply(1:length(SpNetList), function(i) {
>   set.edge.value(SpNetList[[i]], "TRA", value = TRAlist[[i]])
> })
> 
> SpNetList <- lapply(1:length(SpNetList), function(i) {
>   set.edge.value(SpNetList[[i]], "FIN", value = FINlist[[i]])
> })
> 
> ### network dynamic
> SpDy <- networkDynamic(network.list = SpNetList)
> SpDy

**Reply by LettieWyq** [reply]
Could it be the problem that edge attributes were time-varying in my context?
